src/panels/KaitoPanel.ts

```diff
@@ -259,61 +291,74 @@ export class KaitoPanelDataProvider implements PanelDataProvider<"kaito"> {
                 errorMessage: ex instanceof Error ? ex.message : String(ex),
                 models: [],
             });
+
+            return { succeeded: false, error: ex instanceof Error ? ex.message : String(ex) };
         }
+    }
 
-        // install Kaito Federated Credentials and role Assignments
-        try {
-            await longRunning(`Installing Kaito Federated Credentials and role Assignments.`, async () => {
-                const clusterInfo = await getManagedCluster(
-                    this.sessionProvider,
-                    this.subscriptionId,
-                    this.resourceGroupName,
-                    this.clusterName,
-                );
+    private async installKaitoComponents(): Promise<Errorable<string>> {
+        const clusterInfo = await getManagedCluster(
+            this.sessionProvider,
+            this.subscriptionId,
+            this.resourceGroupName,
+            this.clusterName,
+        );
 
-                if (failed(clusterInfo)) {
-                    vscode.window.showErrorMessage(`Error getting managed cluster info: ${clusterInfo.error}`);
-                    return;
-                }
+        if (failed(clusterInfo)) {
+            vscode.window.showErrorMessage(`Error getting managed cluster info: ${clusterInfo.error}`);
+            return { succeeded: false, error: clusterInfo.error };
+        }
 
-                await this.installKaitoRoleAssignments(
-                    clusterInfo.result.nodeResourceGroup!,
-                    this.subscriptionId,
-                    this.resourceGroupName,
-                    this.clusterName,
-                );
+        await this.installKaitoRoleAssignments(
+            clusterInfo.result.nodeResourceGroup!,
+            this.subscriptionId,
+            this.resourceGroupName,
+            this.clusterName,
+        );
 
-                const aksOidcIssuerUrl = clusterInfo.result.oidcIssuerProfile?.issuerURL;
-                if (!aksOidcIssuerUrl) {
-                    vscode.window.showErrorMessage(`Error getting aks oidc issuer url`);
-                    return;
-                }
-                await this.installKaitoFederatedCredentials(
-                    clusterInfo.result.nodeResourceGroup!,
-                    this.clusterName,
-                    aksOidcIssuerUrl,
-                );
+        const aksOidcIssuerUrl = clusterInfo.result.oidcIssuerProfile?.issuerURL;
+        if (!aksOidcIssuerUrl) {
+            vscode.window.showErrorMessage(`Error getting aks oidc issuer url`);
+            return { succeeded: false, error: "Error getting aks oidc issuer url" };
```

Comment:
💡 Thoughts only: Is there any form of `err` above code returns, if yes we can add that in the end of the `error` as extra information if that helps, please.

Reply:
we already handle it in above lines where we retrieve clusterInfo from getManagedClusterCall, this one specifically if the issuerURL is undefined/empty/null after GET managed cluster call - we can finetune the error message to be specific if that's the case 
https://github.com/Azure/vscode-aks-tools/blob/4760b21418889481e7268f23f8a781eabb5c9abf/src/panels/KaitoPanel.ts#L307

I have improved the error message in this PR - https://github.com/tejhan/vscode-aks-tools/pull/2/files